src/Lorentz/Contracts/BaseDAO/Token.hs

```diff
@@ -0,0 +1,75 @@
+-- SPDX-FileCopyrightText: 2020 TQ Tezos
+-- SPDX-License-Identifier: LicenseRef-MIT-TQ
+
+-- | Proposal related entrypoints
+module Lorentz.Contracts.BaseDAO.Token
+  ( burn
+  , mint
+  , transferContractTokens
+  , tokenAddress
+  ) where
+
+import Lorentz
+import Lorentz.Contracts.BaseDAO.Management (authorizeAdmin)
+import Lorentz.Contracts.BaseDAO.Token.FA2 (debitFrom, creditTo)
+import Lorentz.Contracts.BaseDAO.Types
+import qualified Lorentz.Contracts.Spec.FA2Interface as FA2
+
+{-# ANN module ("HLint: ignore Reduce duplication" :: Text) #-}
+
+burn
+  :: forall store pm. StorageC store pm
+  => Entrypoint BurnParam store
+burn = do
+  dip authorizeAdmin
+  swap
+  dip $ do
+    getField #bTokenId
+    dip $ getField #bAmount
+    pair
+    swap
+    toField #bFrom
+
+  debitFrom
+  nil; pair
+
+mint
+  :: forall store pm. StorageC store pm
+  => Entrypoint MintParam store
+mint = do
+  dip authorizeAdmin
+  swap
+  dip $ do
+    getField #mTokenId
+    dip $ getField #mAmount
+    pair
+    swap
+    toField #mTo_
+
+  creditTo
+  nil; pair
+
+transferContractTokens
+  :: forall store pm. StorageC store pm
+  => Entrypoint TransferContractTokensParam store
+transferContractTokens = do
+  getField #tcParams
+  dip $ do
+    toField #tcContractAddress
+    contractCalling @FA2.Parameter (Call @"Transfer")
+    ifSome nop $ do
+      failCustom_ #pROPOSAL_NOT_EXIST -- TODO
+    push zeroMutez
+  transferTokens
+  nil; swap; cons
+  pair
+
+tokenAddress
+  :: forall store pm. (HasAnnotation pm, NiceParameter pm, StorageC store pm)
+  => Entrypoint TokenAddressParam store
+tokenAddress = do
+  -- dip ensureNotMigrated -- TODO
```

Comment:
1. Does it not work?
2. Shouldn't `ensureNotMigrated` be called in other entrypoints?

Reply:
I was mean as a TODO to add `ensureNotMigrated` to all entrypoints except `transferContractTokens`. I believe, it is needed for all entrypoint. Is that correct, Ivan?

Reply:
Yes, it's correct. `transferContractTokens` should still be callable after migration because the contract may have some FA2 tokens after migration (and we don't want to lose them).